### DISCUSSION TITLE
dnsmasq - Persistente DHCP-Leases auch nach Neustart

### DISCUSSION THREAD

**Discussion by wcdgit** [Freetz-NG/freetz-ng#616] [Allgemein]
Gibt es einen Weg DHCP Leases nach einem Neustart der Box nicht zu verlieren? Das in der [Doku](https://freetz-ng.github.io/freetz-ng/make/dnsmasq.html) beschriebene Verhalten: "Mit einen Neustart von dnsmasq werden die Files automatisch neu erstellt." finde ich nicht sehr hilfreich. Da interne (LAN) Geräte nach Neustart der Fritzbox nicht mehr per Hostnamen erreichbar sind.

**Comment by pfichtner** [comment]
> Gibt es einen Weg DHCP Leases nach einem Neustart der Box nicht zu verlieren? 

Du könntest die leases auf einen nicht-flüchtigen Speicher legen, ob dieser sich darüber freut ist allerdings eine andere Frage (und hängt auch stark von der Lease-Time ab). 

> Da interne (LAN) Geräte nach Neustart der Fritzbox nicht mehr per Hostnamen erreichbar sind.

Du scheinst andere Probleme zu haben, das ist unlogisch.

**Reply by wcdgit** [reply]
> > Gibt es einen Weg DHCP Leases nach einem Neustart der Box nicht zu verlieren?
> 
> Du könntest die leases auf einen nicht-flüchtigen Speicher legen, ob dieser sich darüber freut ist allerdings eine andere Frage (und hängt auch stark von der Lease-Time ab).

Ich habe bereits versucht per extra-Option `dhcp-leasefile` das File im Home-Verzeichnis der Box abzulegen.... Leider ohne Erfolg. Nach Neustart der Box wurden keine Leases mehr im Webinterface angezeigt.

Gibt es einen anderen Weg, oder war ich schon auf der richtigen Fährte?

> 
> > Da interne (LAN) Geräte nach Neustart der Fritzbox nicht mehr per Hostnamen erreichbar sind.
> 
> Du scheinst andere Probleme zu haben, das ist unlogisch.

Ich habe keine spezielle Umgebung. Standard freetz-ng auf einer 7490 mit dnsmasq und getdns (stubby).

- Client-A erreicht Server-B ganz normal, solange in den Leases Server-B auftaucht
- Neustart der Fritzbox
- Keine Leases mehr im File
- Client-A kann Server-B nicht mehr auflösen
- DHCP renew auf Server-B
- Client-A kann Server-B wieder erreichen

**Reply by pfichtner** [reply]
> Ich habe bereits versucht per extra-Option `dhcp-leasefile` das File im Home-Verzeichnis der Box abzulegen.... Leider ohne Erfolg. Nach Neustart der Box wurden keine Leases mehr im Webinterface angezeigt.
> 
> Gibt es einen anderen Weg, oder war ich schon auf der richtigen Fährte?

Home ist auch flüchtig, Möglichkeiten wären das ablegen: 
- im NAND, siehe https://freetz.github.io/wiki/help/howtos/development/flash.html
- auf einem USB-Speicher (Stick, HDD, SSD, ...)
- Netzwerkspeicher auf einem weiteren Gerät (smb, nfs, ...)

>     * Client-A erreicht Server-B ganz normal, solange in den Leases Server-B auftaucht
>     * Neustart der Fritzbox
>     * Keine Leases mehr im File
>     * Client-A kann Server-B nicht mehr auflösen
>     * DHCP renew auf Server-B
>     * Client-A kann Server-B wieder erreichen

Ja, das ist so logisch und nachvollziehbar, solange Server-B keinen neuen Lease anfordert/zugewiesen bekommt ist er nicht erreichbar, das sollte der Server-B ja aber eigentlich relativ zeitnah nach der Wiederverfügbarkeit von dnsmasq getan haben. 
Deine Beschreibung hatte ich so interpretiert als ob der Zustand der Nichtereichbarkeit dann bestehen bleibt.

**Reply by wcdgit** [reply]
Kannst du mir einen konkreten Pfad zu einem passenden Ort im Nand nennen/empfehlen? Mein Root-Verzeichnis auf der 7490 sieht folgendermaßen aus:

![grafik](https://user-images.githubusercontent.com/54795276/211148714-0a55d8bf-0df9-4fb6-ac61-4f2bcf283709.png)

Was meintest du eigentlich mit "... ob dieser sich darüber freut ist allerdings eine andere Frage"? Stellt der Schreib-Zugriff ein Problem für den NAND Speicher dar?

**Reply by pfichtner** [reply]
in `var/media/ftp` sind die Mountpoints eingehängter Devices (USB-Sticks, interner Speicher u.U./falls verfügbar). Also dort (je nachdem was Dir vorschwebt). 
Ja, mit der Aussage "freut sich drüber" waren die Schreibvorgänge gemeint. Jeder Speicher verkraftet nur eine bestimmte Anzahl von Schreib/-Löschvorgängen, wie groß die für welchen Speichertyp verrät Dir Google. 

Warum setzt Du nicht einfach die Lease-Time runter? Damit "zwingst" Du die Clients nach Ablauf der Lease-Time einen neuen Lease anzufordern. Wenn Du sie z.B. auf eine Minute runterdrehst, sind Deine Services auch ca. nach dieser Zeit spätestens wieder auflösbar/erreichbar.

**Reply by wcdgit** [reply]
Klasse. Vielen Dank für beide Lösungsansätze. Du kannst [deine Antwort](https://github.com/Freetz-NG/freetz-ng/discussions/616#discussioncomment-4620685) gerne als "answered" markieren.

**Comment by nixmomo** [comment]
Du könntest die Leases auch in die Hosts eintragen
http://fritz.box:81/cgi-bin/file/mod/hosts
`Syntax: <ip> <mac> <interface> <host> [<aliases|#description>]
(z.B.: 10.0.0.1 * * www.local mfh1 # Mein Server) *="nicht definiert"`

dann den haken rein und es sollte auch nach nem Reboot erhalten bleiben
![image](https://user-images.githubusercontent.com/8502506/192993808-7386d14e-6851-4017-9e0b-0113c385626d.png)

**Reply by wcdgit** [reply]
Das wäre auf jeden Fall ein Workaround.
